torch/_dynamo/eval_frame.py

```diff
@@ -737,7 +738,15 @@ def __init__(
                 # Fill node.mata["val"] with faketensor from the input,
                 # if it's not found in matched_input_elements_positions
                 if fake_mode is not None and isinstance(flat_args[i], torch.Tensor):
-                    arg.node.meta["val"] = fake_mode.from_tensor(flat_args[i])
+                    arg.node.meta["val"] = fake_mode.from_tensor(
```

Comment:
Are the constraints also preserved in fake_mode.shape_env, e.g. fake_mode.shape_env.var_to_range? I think `if i in matched_input_elements_to_fake` branch, the constraints are preserved. It would be surprising to find some "val" don't preserve user constraints while some does. It would be great if we can add a test case to make this `constraints and symbols are preserved` invariants  explicit and guarded.

Reply:
good point. do you think it's acceptable for me to leave a TODO here and do it in separate diff? 
Two reasons:
1. I need at least symbols to be preserved to unblock server models.
2. These arguments are not used in the graph therefore the constraints won't be used in meaningful way.

But I agree with you it's a surprise to users.

Reply:
Sounds good as long as it's fixed!